### PR TITLE
Add devpod extension

### PR DESCRIPTION
**Pull request by nicholasq**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/devpod/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# DevPod Changelog
+
+## [Initial Version] - 2024-09-26
```
Comment (pernielsentikaer) [inline extensions/devpod/CHANGELOG.md:3]:
```suggestion

- ## [Initial Version] - {PR_MERGE_DATE}

```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- I didn't have devPod installed but after installing it the path for cli was `/usr/local/bin/devpod` so I was not able to use it I just got below error shown

image

After manually setting the path in the code it worked

image

Raycast07102024-Mae331JX png@2x


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/devpod/CHANGELOG.md (hunk 1 above)
Comment (nicholasq) [reply extensions/devpod/CHANGELOG.md:3]: Thank you for the feedback! I will make some fixes for the devpod binary not being found.

**Comment by pernielsentikaer** [comment]
What does this error mean?

devpod 2024-10-28 at 23 06 36

**Comment by nicholasq** [comment]
> What does this error mean?
> 
> devpod 2024-10-28 at 23 06 36

What were you trying to do? I will test it on my side.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Is there a way so the users doesn't need to input `devPodPath`, there must be some defaults paths we can look in before?

**Comment by nicholasq** [comment]
> Is there a way so the users doesn't need to input `devPodPath`, there must be some defaults paths we can look in before?

There are 3 different possible locations according to the devpod GUI app. I've experimented with trying to detect the correct path with some bash commands in `useExec` but I can't seem to get something working. Do you have any suggestions? If not, I propose we treat this like the other extensions that expect an API key is set before the extension can be used. But in this case, it just needs to be the path to devpod (which I will explain in the extension README).

CleanShot 2024-11-23 at 18 44 37@2x

**Comment by nicholasq** [comment]
I tried pushing up a new commit but I'm getting this error:
```
➜  devpod git:(main) npm run publish

> publish
> npx @raycast/api@latest publish

wait  - validate package.json file ..  /Users/nicholasquezada/Zed/devpod/package.json
   4:11  warning  Extension's title has to be Title Cased. Expected "Devpod"
  15:15  warning  Command's title has to be Title Cased. Expected "Devpod Workspaces"
ready  - validate package.json file
ready  - validate package-lock.json
ready  - validate other lock files
ready  - validate extension icons
ready  - validate extension metadata
ready  - run ESLint
ready  - run Prettier 3.3.3
ready  - getting fork
error  - preparing clone
    WC: Invalid response body while trying to fetch https://api.github.com/search/code?q=%22%5C%22name%5C%22%3A%20%5C%22devpod%5C%22%22%20%22%5C%22author%5C%22%3A%20%5C%22nicholasq%5C%22%22%20repo%3Aray
    cast%2Fextensions%20in%3Afile%20path%3Aextensions%20extension%3Ajson&per_page=3: Premature close
    Code: ERR_STREAM_PREMATURE_CLOSE
```
Is this possibly due to the pull request being stalled?

**Comment by pernielsentikaer** [comment]
Looks like GitHub had a hiccup. Does it work for you now?

**Comment by nicholasq** [comment]
> Looks like GitHub had a hiccup. Does it work for you now?

Silly me. I was using node 18 instead of 20 when I tried to push. Just letting you know in case it helps in the future :)

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**